### ISSUE TITLE
why does it say header-only library when there are cpp files and makefile?

### ISSUE THREAD

**Issue by patefacio** [nekipelov/redisclient#2]
Sorry - never mind - had not actually looked at the code well enough. Interesting setup to allow lib or header only.

**Comment by nekipelov** [comment]
> This call is failing with a protocol error

Can you provide full example?

**Comment by patefacio** [comment]
Unfortunately I had to switch libraries due to this and another issue. The commit that did the switch is: https://github.com/patefacio/RedisExchange/commit/f667b2a07c08a374380e602ed7df18b188149b06

If you get the code prior to that you will have a full example. As I recall, at a minimum there was a timing issue/race condition at startup. If you look at this code https://github.com/patefacio/RedisExchange/blob/8dd3cbf8ca2055bc565b8b6ca1111afbdac5cde9/exch/cpp/app/exch_server/exch_server.cpp#L111-L117 
it is fairly straightforward. But for it to even get through startup correctly I found I needed to throw in a current thread sleep for a second or so before L117. Otherwise it would get some sort of state error. I think the issue is related to there being multiple queues involved and the approach of trying to support pub/sub in the state-machine that looks to cater to connected (i.e. state of NotConnected, Connected, Subscribed, Closed). By combining states of pub/sub with connected, off the bat there was an issue of connecting and subscribing to multiple channels. 

Then with the sleep statement allowing it to get past startup I got an error similar to: https://github.com/nekipelov/redisclient/issues/4
